src/main/java/controllers/ConsoleInputController.java

```diff
@@ -0,0 +1,124 @@
+package controllers;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import models.data.Project;
+import repositories.ProjectRepository;
+import util.log.DukeLogger;
+import views.CLIView;
+
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class ConsoleInputController implements IController {
+
+    private CLIView consoleView;
+    private ProjectRepository projectRepository;
+    private ProjectInputController projectInputController;
+    private String filePath = System.getProperty("user.dir") + "/savedProjects.json";
+
+    /**
+     * Constructor.
+     * @param view : takes in a View model, in this case a command line view.
+     */
+    public ConsoleInputController(CLIView view) {
+        this.consoleView = view;
+        loadProjectsData();
+        this.projectInputController = new ProjectInputController(this.consoleView, this.projectRepository);
```

Comment:
Think of better ways to pass data up or down. `ProjectInputController` should not be aware of `CLIView`. They are separate concerns and should be clearly divided.  

In fact, if `ConsoleInputController` and `ProjectInputController` are implementing the same interface, they should be at the same level of abstraction.